src/snipe/api/multisig_reference_QC.py

```
@@ -766,145 +987,167 @@
+                unique_kmers_array[i] = int(total_reference_kmers * coverage_array[i])
+
+            # We'll define a small helper for the saturation model
             def saturation_model(x, a, b):
-                return a * x / (b + x)
+                # Michaelis–Menten style
+                return (a * x) / (b + x)
+
+            # Pull out x-coords (cumulative abundance), y1=coverage, y2=unique
+            x_data = cumulative_total_abundances
+            y_coverage_data = coverage_array
+            y_unique_data = unique_kmers_array
```

Ruff (0.8.2)

Local variable `y_unique_data` is assigned to but never used

Remove assignment to unused variable `y_unique_data`

(F841)